### PR TITLE
[SYCL][ESIMD][EMU] LSC Atomic update for emulator backend

### PR DESCRIPTION
**Pull request by dongkyunahn-intel**
- Argument order fix for cmpxchg
- Enabling lsc_fence.cpp test

Complementary compiler patch: https://github.com/intel/llvm/pull/6661

### CODE REVIEW

**Comment by dongkyunahn-intel** [comment]
Failures are only on Windows - not related to this patch.

File: SYCL/ESIMD/lsc/Inputs/lsc_surf.hpp [hunk 1]
```diff
@@ -75,8 +75,8 @@ int main() {
             lsc_atomic_update<atomic_op::inc, int>(access_2, offsets, pred);
             lsc_atomic_update<atomic_op::add, int>(access_3, offsets, add,
                                                    pred);
-            lsc_atomic_update<atomic_op::cmpxchg, int>(access_4, offsets,
-                                                       compare, swap, pred);
+            lsc_atomic_update<atomic_op::cmpxchg, int>(access_4, offsets, swap,
```
Comment (kbobrovs) [inline SYCL/ESIMD/lsc/Inputs/lsc_surf.hpp]: I wonder how this worked before...

File: SYCL/ESIMD/lsc/lsc_slm.cpp [hunk 2]
```diff
@@ -81,8 +81,8 @@ int main() {
             lsc_block_store<int, SIMDSize>(access_3, offset, data_3);
 
             lsc_slm_block_store<int, SIMDSize>(offset, data);
-            lsc_slm_atomic_update<atomic_op::cmpxchg, int>(offsets, compare,
-                                                           swap, pred);
+            lsc_slm_atomic_update<atomic_op::cmpxchg, int>(offsets, swap,
```
Comment (kbobrovs) [inline SYCL/ESIMD/lsc/lsc_slm.cpp]: please check these change on PVC

Comment (dongkyunahn-intel) [reply SYCL/ESIMD/lsc/lsc_slm.cpp]: Reverted.

**Review by kbobrovs — APPROVED** [review]

**Pull request merged**